src/components/AddressSearch/index.js

```diff
@@ -122,6 +143,19 @@ function AddressSearch(props) {
     const saveLocationDetails = (autocompleteData, details) => {
         const addressComponents = details.address_components;
         if (!addressComponents) {
+            // When there are details, but no address_components, this indicates that some predefined options have been passed
+            // to this component which don't match the usual properties coming from auto-complete. In that case, only a limited
+            // amount of data massaging needs to happen for what the parent expects to get from this function.
+            if (_.size(details)) {
+                props.onPress({
+                    address: lodashGet(details, 'description', ''),
+                    lat: lodashGet(details, 'geometry.location.lat', 0),
+                    lng: lodashGet(details, 'geometry.location.lng', 0),
+                });
+
+                // After we select an option, we set displayListViewBorder to false to prevent UI flickering
+                setDisplayListViewBorder(false);
```

Comment:
This should be outside the if condition. I think we should actually revert the change related to this line. It's better to just call this function on the `onPress` callback once rather then calling it before every return statement in `saveLocationDetails`

Reply:
Sure, I can move it back outside. It made more sense to move it inside before I added this block and had to repeat calling `setDisplayListViewBorder` in two locations.